3scaleAdapter/pkg/threescale/threescale.go

```diff
@@ -273,10 +292,32 @@ func NewThreescale(addr string, client *http.Client, proxyCache *ProxyConfigCach
 		proxyCache: proxyCache,
 	}
 
+	if metricsConf != nil {
+		if metricsConf.gatherMetrics {
+			s.reportMetrics = true
+		}
+
+		if proxyCache != nil {
+			proxyCache.reportMetrics = true
+		}
+
+		http.Handle("/metrics", promhttp.Handler())
+		go func() {
+			if err := http.ListenAndServe(fmt.Sprintf(":%d", metricsConf.port), nil); err != nil {
+				panic("panic")
+			}
+		}()
+	}
+
 	log.Infof("Threescale Istio Adapter is listening on \"%v\"\n", s.Addr())
 
 	s.server = grpc.NewServer()
 	authorization.RegisterHandleAuthorizationServiceServer(s.server, s)
 	// TODO: Add report template for metrics.
 	return s, nil
 }
+
+// NewMetricsConfig - Creates a MetricsConfig which controls metric behaviour
+func NewMetricsConfig(gatherMetrics bool, port int) *MetricsConfig {
+	return &MetricsConfig{gatherMetrics, port}
+}
```

Comment:
I'd split this into its own (perhaps tiny) file - there you can define the default port and keep it private:

This is so that you:
- avoid coupling external code to the definition of default port currently defined and used elsewhere
- provide an interface in which you can specify a valid port or some sort of value meant to be "default", that would then return the default port as determined by this module

In this specific instance I am not sure of how useful the `gatherMetrics` boolean is - I assume it could be useful if you could enable/disable it dynamically?

Reply:
Ok sure. Well it can be toggled from an environment variable. In the linked issue I mentioned that the labels for these metrics are unbounded so I think that flow makes sense?

Reply:
update: well, on a second thought I think it is ok as it is - I'd just specify 0 or a negative value as "use the default port", but it is still some sort of coupling equivalent to just using a DefaultPort constant (just to clarify, the constant would still belong in this tiny module)

Reply:
Ok think I've resolved this one now